Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -1991,7 +1991,9 @@ extension URL {
 
 #if !NO_FILESYSTEM
     private static func isDirectory(_ path: String) -> Bool {
-        return FileManager.default._fileStat(path)?.isDirectory ?? false
+        var isDirectory: Bool = false
+        FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
```

Comment:
I'm seeing a warning that states:

```
Result of call to 'fileExists(atPath:isDirectory:)' is unused
```

here, any chance we can add a `_ =` to silence that warning?

Also, when building this change as part of the internal Foundation framework, we hit the error

```
Sources/FoundationEssentials/URL/URL.swift:1995:67: error: cannot convert value of type 'UnsafeMutablePointer<Bool>' to expected argument type 'UnsafeMutablePointer<ObjCBool>'
```

Because on Darwin this API uses an `ObjCBool` instead of a `Bool`. Could we update it to account for that, maybe something like

```swift
#if !FOUNDATION_FRAMEWORK
var isDirectory: Bool = false
_ = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
return isDirectory
#else
var isDirectory: ObjCBool = false
_ = FileManager.default.fileExists(atPath: path, isDirectory: &isDirectory)
return isDirectory.boolValue
#endif
```

We probably need to rethink the portability of this function but for now that should let us merge this change

Reply:
Trying a small tweak to see if we can just rely on the type inference to figure out the types.

Reply:
Sadly it's not smart enough to infer the type of `isDirectory` from the usage on the line below 😢:

```
Sources/FoundationEssentials/URL/URL.swift:1995:71: error: cannot convert value of type 'UnsafeMutablePointer<Bool>' to expected argument type 'UnsafeMutablePointer<ObjCBool>'
Sources/FoundationEssentials/URL/URL.swift:1999:28: error: value of type 'Bool' has no member 'boolValue'
```